test/test_profiler.py

```diff
@@ -303,6 +306,35 @@ def test_export_stacks(self):
                     pass
                 assert is_int, "Invalid stacks record"
 
+    @unittest.skipIf(not kineto_available(), "Kineto is required")
+    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is required")
+    def test_tensorboard_trace_handler(self):
+        with profile(use_cuda=True, use_kineto=True):
+            self.payload()
+
+        tempdir = os.path.join(tempfile.gettempdir(), "tensorboard_logdir{}".format(str(int(time.time() * 1000))))
+        with torch.profiler.profile(
+            activities=[
+                torch.profiler.ProfilerActivity.CPU,
+                torch.profiler.ProfilerActivity.CUDA],
+            schedule=torch.profiler.schedule(
+                wait=1,
+                warmup=1,
+                active=2),
+            on_trace_ready=torch.profiler.tensorboard_trace_handler(tempdir)
+        ) as p:
+            for _ in range(8):
+                self.payload()
+                p.next_step()
+
+        assert os.path.exists(tempdir)
+        file_num = 0
+        for file_name in os.listdir(tempdir):
+            assert int(file_name.split('.')[-4]) > 0, "Wrong tracing file name pattern"
+            self.assertEqual(file_name.split('.')[-3:], ['pt', 'trace', 'json'])
+            file_num += 1
+        self.assertEqual(file_num, 2)
+
```

Comment:
Should be delete the tempdir, or at least files under it?
Keeping it is also a choice, for checking file content after test.